stan/math/gpu/copy_triangular_transposed_opencl.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef STAN_MATH_GPU_COPY_TRIANGULAR_TRANSPOSED_OPENCL_HPP
+#define STAN_MATH_GPU_COPY_TRIANGULAR_TRANSPOSED_OPENCL_HPP
+#ifdef STAN_OPENCL
+#include <stan/math/gpu/matrix_gpu.hpp>
+#include <stan/math/gpu/err/check_square.hpp>
+#include <CL/cl.hpp>
+
+namespace stan {
+namespace math {
+enum copy_transposed_triangular {
```

Comment:
This name is just confusing. Is it meant to be used outside of this function?

Is an enum necessary?

Reply:
The enum is a 'nicety' we could instead just have the user pass a number to indicate how they want the transpose done